src/rpm/headers/types.rs

```diff
@@ -505,4 +652,53 @@ mod test {
         blank_file.caps("cap_net_an,cap_net_raw+p").unwrap_err();
         Ok(())
     }
+
+    #[test]
+    fn test_scriptlet_builder() {
+        // Test full state
+        let _scriptlet = crate::Scriptlet::new(
+                r#"
+echo `hello world`
+        "#
+                .trim(),
+            )
+            .flags(crate::ScriptletFlags::EXPAND)
+            .prog(vec!["/usr/bin/blah", "-c"])
+            .ty(crate::ScriptletType::PreInstall);
```

Comment:
Since we're passing it straight into `.pre_install_script()` or what have you, there's no real need to specify the type independently